smaps/src/test/java/com/google/smaps/FileParserTest.java

```
@@ -0,0 +1,127 @@
+/**
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.smaps;
+
+import static org.junit.Assert.*;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+/**
+ * Unit tests for {@link FileParser}.
+ */
+@RunWith(JUnit4.class)
+public class FileParserTest {
+  private List<Region> regions;
+
+  @Before
+  public void setUp() throws Exception {
+    // Creates regions list from smaps-full.txt file.
+    String filePathname = "../smaps-full.txt";
+    regions = FileParser.getRegionList(filePathname);
+  }
+
+  @Test
+  public void fileNotFound() {
+    // Tests that an empty list is successfully returned with nonexistent file.
+    String nonexistentFilePathname = "../fake-file.txt";
+    List<Region> list = FileParser.getRegionList(nonexistentFilePathname);
+    assertEquals(0, list.size());
+  }
+
+  @Test
+  public void wrongFileFormat() {
```

This is totally fine for now, but we may want to move to throwing exceptions for formats that we can't handle instead of returning lists of 0. We will have to think about how we want the visualizer to react when a user uploads something incorrect. No action item here for now, just something to think about :)